apps/etterlatte-beregning/src/main/kotlin/BeregningRepository.kt

```diff
@@ -0,0 +1,142 @@
+package no.nav.etterlatte
+
+import com.fasterxml.jackson.module.kotlin.readValue
+import io.ktor.server.plugins.NotFoundException
+import kotliquery.Row
+import kotliquery.TransactionalSession
+import kotliquery.queryOf
+import kotliquery.sessionOf
+import kotliquery.using
+import no.nav.etterlatte.libs.common.beregning.Beregningsperiode
+import no.nav.etterlatte.libs.common.grunnlag.Metadata
+import no.nav.etterlatte.libs.common.objectMapper
+import no.nav.etterlatte.libs.common.toJson
+import no.nav.etterlatte.model.Beregning
+import java.util.*
+import javax.sql.DataSource
+
+interface BeregningRepository {
+    fun lagre(beregning: Beregning): Beregning
+    fun hent(behandlingId: UUID): Beregning
+}
+
+class BeregningRepositoryImpl(private val dataSource: DataSource) : BeregningRepository {
+    override fun lagre(
+        beregning: Beregning
+    ): Beregning {
+        using(sessionOf(dataSource)) { session ->
+            session.transaction { tx ->
+                {
+                    queryOf(
+                        statement = Queries.lagreBeregning,
+                        paramMap = mapOf(
+                            "beregningId" to beregning.beregningId,
+                            "behandlingId" to beregning.behandlingId,
+                            "beregnetDato" to beregning.beregnetDato,
+                            "beregningsperioder" to beregning.beregningsperioder.toJson(),
+                            "sakId" to beregning.grunnlagMetadata.sakId,
+                            "grunnlagVersjon" to beregning.grunnlagMetadata.versjon
+                        )
+                    ).let { query -> tx.run(query.asUpdate) }
+
+                    beregning.beregningsperioder.forEach {
+                        queryOf(
+                            statement = Queries.lagreBeregningPeriode,
+                            paramMap = mapOf(
+                                "beregningId" to beregning.beregningId,
+                                "datoFOM" to it.datoFOM,
+                                "datoTOM" to it.datoTOM,
+                                "utbetaltBeloep" to it.utbetaltBeloep,
+                                "soeskenFlokk" to it.soeskenFlokk,
+                                "grunnbeloepMnd" to it.grunnbelopMnd,
+                                "grunnbeloep" to it.grunnbelopMnd
+                            )
+                        ).let { query -> tx.run(query.asUpdate) }
+                    }
+                }
+            }
```

Comment:
Lagres både beregning med beregningsperioder, og beregningsperioder for sig selv i tillegg? håller det ikke med den ene bare? 😄

Reply:
Da er spørsmålet hva en beregning er, en beregning består jo av mange perioder. Da vil vi ikke få en key som identifiserer alle perioder så det er en avveining på det :D

Reply:
Godt innspill! Man hadde faktisk klart seg med bare beregningsperioder (man kunne laget en kolonne for beregningsid som samler de på tvers)